File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/common/uicomponent/gift/giftcloudserver/IGiftCloudServer.java
package com.trtc.uikit.livekit.common.uicomponent.gift.giftcloudserver;

import com.trtc.uikit.livekit.common.uicomponent.gift.model.TUIGift;

import java.util.List;

public interface IGiftCloudServer {

    void rechargeBalance(Callback<Integer> callback);

    void queryBalance(Callback<Integer> callback);

    void queryGiftInfoList(Callback<List<TUIGift>> callback);

    void sendGift(String sender, String receiver, TUIGift gift, int giftCount, Callback<Integer> callback);

    interface Callback<T> {
        void onResult(int error, T result);
    }

    interface Error {
        int NO_ERROR = 0;
        int OPERATION_FAILED = -1;
        int PARAM_ERROR = -2;
        int BALANCE_INSUFFICIENT = -3;
    }
}

File: iOS/TUILiveKit/Sources/Component/GiftAccess/GiftCloudServer/IGiftCloudServer.swift
//
//  IGiftCloudServer.swift
//  TUILiveKit
//
//  Created by krabyu on 2024/4/3.
//

import Foundation
import TUILiveComponent

protocol IGiftCloudServer {
    func rechargeBalance(callback: @escaping (TUIGiftServerError, Int) -> Void)
    func queryBalance(callback: @escaping (TUIGiftServerError, Int) -> Void)
    func queryGiftInfoList(callback: @escaping (TUIGiftServerError, [TUIGift]) -> Void)
    func sendGift(sender: String, receiver: String,
                  giftModel: TUIGift, giftCount: Int,
                  callback: @escaping (TUIGiftServerError, Int) -> Void)
}

enum TUIGiftServerError: Int {
    case noError = 0
    case operationFailed = -1
    case paramError = -2
    case balanceInsufficient = -3
}
